var/spack/repos/builtin/packages/libpng/package.py

```diff
@@ -38,12 +40,14 @@ class Libpng(CMakePackage):
         description="Build shared libs, static libs or both",
     )
 
-
 class CMakeBuilder(CMakeBuilder):
     def cmake_args(self):
-        return [
+        args = [
             self.define("CMAKE_CXX_FLAGS", self.spec["zlib"].headers.include_flags),
             self.define("ZLIB_ROOT", self.spec["zlib"].prefix),
             self.define("PNG_SHARED", "shared" in self.spec.variants["libs"].value),
             self.define("PNG_STATIC", "static" in self.spec.variants["libs"].value),
-        ]
+            ]
+        if platform.system() == "Darwin" and platform.machine() == "arm64":
```

Comment:
Can you try
```suggestion
        if self.spec.satisfies("platform=darwin target=aarch64:"):
```
and then revert the `import platform` changes in lines 6-7 please?

Reply:
Your suggestion worked. Thanks! I think it's an improvement since it doesn't rely on the python platform module.